--- vnc-4_1_1-unixsrc/unix/xorg-server-1.0.0/configure.ac.fontpath	2005-12-21 11:14:58.000000000 +0000
+++ vnc-4_1_1-unixsrc/unix/xorg-server-1.0.0/configure.ac	2005-12-21 11:16:56.000000000 +0000
@@ -339,9 +339,9 @@
 AC_ARG_WITH(mesa-source,     AS_HELP_STRING([--with-mesa-source=MESA_SOURCE], [Path to Mesa source tree]),
                                 [ MESA_SOURCE="$withval" ],
                                 [ MESA_SOURCE="" ])
-AC_ARG_WITH(fontdir,         AS_HELP_STRING([--with-fontdir=FONTDIR], [Path to top level dir where fonts are installed (default: ${libdir}/X11/fonts)]),
+AC_ARG_WITH(fontdir,         AS_HELP_STRING([--with-fontdir=FONTDIR], [Path to top level dir where fonts are installed (default: ${datadir}/X11/fonts)]),
                                 [ FONTDIR="$withval" ],
-                                [ FONTDIR="${libdir}/X11/fonts" ])
+                                [ FONTDIR="${datadir}/X11/fonts" ])
 DEFAULT_FONT_PATH="${FONTDIR}/misc/,${FONTDIR}/TTF/,${FONTDIR}/OTF,${FONTDIR}/Type1/,${FONTDIR}/CID/,${FONTDIR}/100dpi/,${FONTDIR}/75dpi/"
 AC_ARG_WITH(default-font-path, AS_HELP_STRING([--with-default-font-path=PATH], [Comma separated list of font dirs]),
 				[ FONTPATH="$withval" ],
