--- vnc-3.3.6-unixsrc/vncserver~	Wed Oct 30 11:21:33 2002
+++ vnc-3.3.6-unixsrc/vncserver	Sun Feb  2 21:13:07 2003
@@ -43,10 +43,7 @@
 
 $defaultXStartup
     = ("#!/bin/sh\n\n".
-       "[ -r \$HOME/.Xresources ] && xrdb \$HOME/.Xresources\n".
-       "xsetroot -solid grey\n".
-       "xterm -geometry 80x24+10+10 -ls -title \"\$VNCDESKTOP Desktop\" &\n".
-       "twm &\n");
+       "exec /etc/X11/xinit/xinitrc &\n");
 
 chop($host = `uname -n`);
 
