--- vnc_unixsrc/vncconnect/vncconnect.c.typo	Mon Nov  6 13:01:09 2000
+++ vnc_unixsrc/vncconnect/vncconnect.c	Mon Nov  6 13:01:33 2000
@@ -11,7 +11,7 @@
 static void usage()
 {
     fprintf(stderr, "usage: %s [-display Xvnc-display] host[:port]\n"
-	    "Tells Xvnc to connects to a listening VNC viewer on the given"
+	    "Tells Xvnc to connect to a listening VNC viewer on the given"
 							    " host and port\n",
 	    programName);
     exit(1);
