--- vnc-4_1_1-unixsrc/unix/x0vncserver/x0vncserver.cxx.select	2005-12-05 10:02:30.000000000 +0000
+++ vnc-4_1_1-unixsrc/unix/x0vncserver/x0vncserver.cxx	2005-12-05 10:02:32.000000000 +0000
@@ -302,6 +302,7 @@
       fd_set rfds;
       std::list<Socket*> sockets;
       std::list<Socket*>::iterator i;
+      tv.tv_usec = tv.tv_sec = 0;
 
       // Process any incoming X events
       TXWindow::handleXEvents(dpy);
