--- vnc-4_1_1-unixsrc/unix/xorg-server-1.1.0/os/access.c.null-interface	2005-11-08 07:33:30.000000000 +0100
+++ vnc-4_1_1-unixsrc/unix/xorg-server-1.1.0/os/access.c	2006-05-09 09:36:25.000000000 +0200
@@ -975,12 +975,14 @@
 	return;
     }
     for (ifr = ifap; ifr != NULL; ifr = ifr->ifa_next) {
+	  if (!ifr->ifa_addr)
+		continue;
 #ifdef DNETCONN
 	if (ifr->ifa_addr.sa_family == AF_DECnet) 
 	    continue;
 #endif /* DNETCONN */
 	len = sizeof(*(ifr->ifa_addr));
-	family = ConvertAddr(ifr->ifa_addr, &len, (pointer *)&addr);
+	family = ifr->ifa_addr ? ConvertAddr(ifr->ifa_addr, &len, (pointer *)&addr) : -1;
 	if (family == -1 || family == FamilyLocal) 
 	    continue;
 #if defined(IPv6) && defined(AF_INET6)
