--- vnc-4_1-unixsrc/unix/xc/programs/Xserver/vnc/vncExtInit.cc.gcc4	2005-03-04 13:52:03.000000000 +0000
+++ vnc-4_1-unixsrc/unix/xc/programs/Xserver/vnc/vncExtInit.cc	2005-03-04 13:52:24.000000000 +0000
@@ -751,7 +751,7 @@
   rep.timeout = qcTimeout;
   rep.addrLen = qcTimeout ? strlen(qcAddress) : 0;
   rep.userLen = qcTimeout ? strlen(qcUsername) : 0;
-  rep.opaqueId = (CARD32)queryConnectId;
+  rep.opaqueId = (CARD32)(long)queryConnectId;
   rep.length = (rep.userLen + rep.addrLen + 3) >> 2;
   if (client->swapped) {
     swaps(&rep.sequenceNumber, n);
