--- vnc-4_1_2-unixsrc/unix/xc/programs/Xserver/vnc/Xvnc/xvnc.cc.modular-xorg	2006-06-13 17:47:02.000000000 -0400
+++ vnc-4_1_2-unixsrc/unix/xc/programs/Xserver/vnc/Xvnc/xvnc.cc	2006-06-13 17:47:02.000000000 -0400
@@ -36,8 +36,6 @@
 extern "C" {
 #define class c_class
 #define public c_public
-#define xor c_xor
-#define and c_and
 #ifdef WIN32
 #include <X11/Xwinsock.h>
 #endif
@@ -72,11 +70,8 @@
 #include "inputstr.h"
 #include "keysym.h"
   extern int defaultColorVisualClass;
-  extern char buildtime[];
 #undef class
 #undef public
-#undef xor
-#undef and
 #ifndef VNC_USE_FB
   extern Bool cfb16ScreenInit(ScreenPtr, pointer, int, int, int, int, int);
   extern Bool cfb32ScreenInit(ScreenPtr, pointer, int, int, int, int, int);
@@ -221,7 +216,7 @@
 
   void ddxUseMsg()
   {
-    ErrorF("\nXvnc %s - built %s\n%s", XVNCVERSION, buildtime, XVNCCOPYRIGHT);
+    ErrorF("\nXvnc %s\n%s", XVNCVERSION, XVNCCOPYRIGHT);
     ErrorF("Underlying X server release %d, %s\n\n", VENDOR_RELEASE,
            VENDOR_STRING);
     ErrorF("-screen scrn WxHxD     set screen's width, height, depth\n");
@@ -927,7 +922,7 @@
 
 void InitOutput(ScreenInfo *screenInfo, int argc, char **argv)
 {
-  ErrorF("\nXvnc %s - built %s\n%s", XVNCVERSION, buildtime, XVNCCOPYRIGHT);
+  ErrorF("\nXvnc %s\n%s", XVNCVERSION, XVNCCOPYRIGHT);
   ErrorF("Underlying X server release %d, %s\n\n", VENDOR_RELEASE,
          VENDOR_STRING);
   wellKnownSocketsCreated = true;
--- /dev/null	2006-06-13 10:12:45.634275750 -0400
+++ vnc-4_1_2-unixsrc/unix/xorg-server/hw/vnc/Makefile.am	2006-06-13 17:47:02.000000000 -0400
@@ -0,0 +1,85 @@
+noinst_LTLIBRARIES = libvnccommon.la
+libvnccommon_la_SOURCES = \
+		RegionHelper.h     \
+		vncExtInit.cc     \
+		vncExtInit.h      \
+		vncHooks.cc       \
+		vncHooks.h        \
+		XserverDesktop.cc \
+		XserverDesktop.h
+
+libvnccommon_la_CXXFLAGS = \
+		-DVENDOR_RELEASE="$(VENDOR_RELEASE)" \
+		-DVENDOR_STRING="\"$(VENDOR_STRING)\"" \
+		-DGC_HAS_COMPOSITE_CLIP \
+		-DVNC_USE_FB \
+		-I/usr/include/X11 \
+		-I../../../../common \
+		-I../../../vncconfig
+
+bin_PROGRAMS = Xvnc
+
+Xvnc_SOURCES = \
+		xvnc.cc \
+		$(top_srcdir)/Xext/dpmsstubs.c \
+		$(top_srcdir)/Xi/stubs.c \
+		$(top_srcdir)/mi/miinitext.c \
+		$(top_srcdir)/fb/fbcmap.c
+
+Xvnc_LDADD =	$(XORG_CORE_LIBS) \
+		$(XORG_LIBS) \
+		$(XSERVER_LIBS) \
+		../../fb/libfb.la \
+		../../mi/libminimi.la \
+		../../render/librender.la \
+		../../xkb/libxkbstubs.la \
+		../../dbe/libdbe.la \
+		../../Xext/libXext.la \
+		../../XTrap/libxtrap.la \
+		../../record/librecord.la \
+		../xfree86/os-support/libxorgos.la \
+		../../os/libos.la \
+		../xfree86/common/libcommon.la \
+		-lX11 \
+		libvnccommon.la \
+		../../../../common/rfb/librfb.a \
+		../../../../common/rdr/librdr.a \
+		../../../../common/network/libnetwork.a \
+		../../../../common/Xregion/libXregion.a
+
+Xvnc_CFLAGS =  -DHAVE_DIX_CONFIG_H \
+		-DNO_HW_ONLY_EXTS \
+		-DNO_MODULE_EXTS \
+		-DXFree86Server -DVNCEXT
+
+Xvnc_CXXFLAGS = $(Xvnc_CFLAGS) \
+		-DVENDOR_RELEASE="$(VENDOR_RELEASE)" \
+		-DVENDOR_STRING="\"$(VENDOR_STRING)\"" \
+		-DVNC_USE_FB \
+		-I../../../../common \
+		-I../../../vncconfig \
+		-I../../mi \
+		-I ../../render \
+		-I/usr/include/X11
+
+libvnc_la_LTLIBRARIES = libvnc.la
+libvnc_la_LDFLAGS = -module -avoid-version
+libvnc_ladir = $(moduledir)/extensions
+libvnc_la_SOURCES = xf86vncModule.cc
+libvnc_la_LIBADD = \
+		libvnccommon.la \
+		../../../../common/rfb/librfb.a \
+		../../../../common/rdr/librdr.a \
+		../../../../common/network/libnetwork.a \
+		../../../../common/Xregion/libXregion.a
+libvnc_la_CXXFLAGS = \
+		-I../../../../common \
+		-I$(top_srcdir)/hw/xfree86/common \
+		-I$(top_srcdir)/hw/xfree86/os-support \
+		-I$(top_srcdir)/hw/xfree86/os-support/bus \
+		-DXFree86Module -DXFree86LOADER -DIN_MODULE
+
+AM_CXXFLAGS =	$(CFLAGS) \
+		-DVENDOR_RELEASE="$(VENDOR_RELEASE)" \
+		-DVENDOR_STRING="\"$(VENDOR_STRING)\""
+
--- vnc-4_1_2-unixsrc/unix/xorg-server/hw/Makefile.am.modular-xorg	2006-01-09 21:30:56.000000000 -0500
+++ vnc-4_1_2-unixsrc/unix/xorg-server/hw/Makefile.am	2006-06-13 17:47:51.000000000 -0400
@@ -42,6 +42,8 @@
 DARWIN_SUBDIRS = darwin
 endif
 
+XVNC_SUBDIRS = vnc
+
 SUBDIRS =			\
 	$(XORG_SUBDIRS)		\
 	$(XGL_SUBDIRS)		\
@@ -35,6 +37,7 @@
 	$(XWIN_SUBDIRS)		\
 	$(XVFB_SUBDIRS)		\
 	$(XNEST_SUBDIRS)	\
+	$(XVNC_SUBDIRS)	\
 	$(DMX_SUBDIRS)          \
         $(KDRIVE_SUBDIRS)	\
 	$(XPRINT_SUBDIRS)
--- vnc-4_1_2-unixsrc/unix/xorg-server/configure.ac~	2008-01-08 03:18:32.000000000 +0200
+++ vnc-4_1_2-unixsrc/unix/xorg-server/configure.ac	2008-01-08 03:21:00.938517181 +0200
@@ -2096,6 +2096,7 @@
 hw/darwin/iokit/Makefile
 hw/darwin/quartz/Makefile
 hw/darwin/utils/Makefile
+hw/vnc/Makefile
 hw/kdrive/Makefile
 hw/kdrive/ati/Makefile
 hw/kdrive/chips/Makefile
--- vnc-4_1_2-unixsrc/unix/xorg-server/mi/miinitext.c.modular-xorg	2006-03-11 19:11:34.000000000 -0500
+++ vnc-4_1_2-unixsrc/unix/xorg-server/mi/miinitext.c	2006-06-13 17:47:02.000000000 -0400
@@ -297,6 +297,9 @@
 #ifdef MITMISC
 extern void MITMiscExtensionInit(INITARGS);
 #endif
+#ifdef VNCEXT
+extern void vncExtensionInit(INITARGS);
+#endif
 #ifdef XIDLE
 extern void XIdleExtensionInit(INITARGS);
 #endif
@@ -577,6 +580,9 @@
 #ifdef MITMISC
     if (!noMITMiscExtension) MITMiscExtensionInit();
 #endif
+#ifdef VNCEXT
+    vncExtensionInit();
+#endif
 #ifdef XIDLE
     if (!noXIdleExtension) XIdleExtensionInit();
 #endif
--- vnc-4_1_2-unixsrc/unix/vncinstall.modular-xorg	2005-02-28 08:00:32.000000000 -0500
+++ vnc-4_1_2-unixsrc/unix/vncinstall	2006-06-13 17:47:02.000000000 -0400
@@ -56,7 +56,7 @@
   fi
 fi
 
-for f in xc/programs/Xserver/Xvnc vncviewer/vncviewer vncpasswd/vncpasswd \
+for f in xorg-server-*/hw/vnc/Xvnc vncviewer/vncviewer vncpasswd/vncpasswd \
 	 vncconfig/vncconfig vncserver x0vncserver/x0vncserver; do
   if [ ! -f $f ]; then
     echo "Couldn't find $f"
@@ -85,8 +85,8 @@
 
 done
 
-vncModule=xc/programs/Xserver/vnc/module/vnc.so
-if [ -f "$vncModule" -a -d "$moduledst" ]; then
+vncModule=xorg-server-*/hw/vnc/.libs/libvnc.so
+if [ -f $vncModule -a -d "$moduledst" ]; then
   if cmp -s $vncModule $moduledst/`basename $vncModule`; then
     echo "`basename $vncModule` hasn't changed"
   else
