--- vnc-4_1-unixsrc/unix/xc/programs/Xserver/vnc/module/Imakefile.fPIC	2005-03-04 14:33:47.000000000 +0000
+++ vnc-4_1-unixsrc/unix/xc/programs/Xserver/vnc/module/Imakefile	2005-03-04 14:34:16.000000000 +0000
@@ -21,7 +21,8 @@
 LinkSourceFile(xf86vncModule.cc,..)
 LinkSourceFile(XserverDesktop.cc,..)
 
-ModuleObjectRule()
+.CCsuf.Osuf:
+	NormalSharedLibObjCplusplusCompile($(_NOOP_))
 /*
  LibraryModuleTarget(vnc,$(OBJS) $(VNCLIBS))
  InstallLibraryModule(vnc,$(MODULEDIR),extensions)
--- vnc-4_1-unixsrc/unix/xc/programs/Xserver/vnc/Imakefile.fPIC	2005-03-04 14:34:24.000000000 +0000
+++ vnc-4_1-unixsrc/unix/xc/programs/Xserver/vnc/Imakefile	2005-03-04 14:34:52.000000000 +0000
@@ -36,8 +36,8 @@
 NormalLintTarget($(SRCS))
 
 NormalLibraryObjectRule()
-NormalCplusplusObjectRule()
-
+.CCsuf.Osuf:
+	NormalSharedLibObjCplusplusCompile($(_NOOP_))
 
 MakeSubdirs($(SUBDIRS))
 DependSubdirs($(SUBDIRS))
--- vnc-4_1-unixsrc/unix/boilerplate.mk.fPIC	2005-03-04 14:35:03.000000000 +0000
+++ vnc-4_1-unixsrc/unix/boilerplate.mk	2005-03-04 14:35:23.000000000 +0000
@@ -15,7 +15,7 @@
 CFLAGS = @CFLAGS@ $(DIR_CFLAGS)
 CCLD = $(CC)
 CXX = @CXX@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = @CXXFLAGS@ $(DIR_CXXFLAGS)
 CXXLD = $(CXX)
 CPPFLAGS = @CPPFLAGS@
 DEFS = @DEFS@
--- vnc-4_1-unixsrc/common/Xregion/Makefile.in.fPIC	2005-03-04 14:32:24.000000000 +0000
+++ vnc-4_1-unixsrc/common/Xregion/Makefile.in	2005-03-04 14:32:36.000000000 +0000
@@ -12,4 +12,5 @@
 	$(AR) $(library) $(OBJS)
 	$(RANLIB) $(library)
 
+DIR_CFLAGS = -DPIC -fPIC
 # followed by boilerplate.mk
--- vnc-4_1-unixsrc/common/network/Makefile.in.fPIC	2005-03-04 14:32:44.000000000 +0000
+++ vnc-4_1-unixsrc/common/network/Makefile.in	2005-03-04 14:32:58.000000000 +0000
@@ -14,4 +14,5 @@
 	$(AR) $(library) $(OBJS)
 	$(RANLIB) $(library)
 
+DIR_CXXFLAGS = -DPIC -fPIC
 # followed by boilerplate.mk
--- vnc-4_1-unixsrc/common/rdr/Makefile.in.fPIC	2005-03-04 14:33:05.000000000 +0000
+++ vnc-4_1-unixsrc/common/rdr/Makefile.in	2005-03-04 14:33:16.000000000 +0000
@@ -16,4 +16,5 @@
 	$(AR) $(library) $(OBJS)
 	$(RANLIB) $(library)
 
+DIR_CXXFLAGS = -DPIC -fPIC
 # followed by boilerplate.mk
--- vnc-4_1-unixsrc/common/rfb/Makefile.in.fPIC	2005-03-04 14:33:20.000000000 +0000
+++ vnc-4_1-unixsrc/common/rfb/Makefile.in	2005-03-04 14:33:33.000000000 +0000
@@ -65,4 +65,6 @@
 	$(AR) $(library) $(OBJS)
 	$(RANLIB) $(library)
 
+DIR_CFLAGS = -DPIC -fPIC
+DIR_CXXFLAGS = -DPIC -fPIC
 # followed by boilerplate.mk
--- vnc-4_1-unixsrc/common/Makefile.in.fPIC	2005-03-04 15:08:57.000000000 +0000
+++ vnc-4_1-unixsrc/common/Makefile.in	2005-03-04 15:09:25.000000000 +0000
@@ -1,4 +1,6 @@
 
 SUBDIRS = @ZLIB_DIR@ rdr network Xregion rfb
 
+DIR_CFLAGS = -DPIC -fPIC
+DIR_CXXFLAGS = -DPIC -fPIC
 # followed by boilerplate.mk
--- vnc-4_1-unixsrc/common/boilerplate.mk.fPIC	2005-03-04 15:09:32.000000000 +0000
+++ vnc-4_1-unixsrc/common/boilerplate.mk	2005-03-04 15:09:49.000000000 +0000
@@ -15,7 +15,7 @@
 CFLAGS = @CFLAGS@ $(DIR_CFLAGS)
 CCLD = $(CC)
 CXX = @CXX@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = @CXXFLAGS@ $(DIR_CXXFLAGS)
 CXXLD = $(CXX)
 CPPFLAGS = @CPPFLAGS@
 DEFS = @DEFS@
