diff -Nur vnc-3.3.7-unixsrc.orig/Xvnc/config/cf/Imake.cf vnc-3.3.7-unixsrc/Xvnc/config/cf/Imake.cf
--- vnc-3.3.7-unixsrc.orig/Xvnc/config/cf/Imake.cf	2002-05-11 12:57:05.000000000 +0200
+++ vnc-3.3.7-unixsrc/Xvnc/config/cf/Imake.cf	2004-08-22 11:07:39.811947966 +0200
@@ -545,17 +545,227 @@
 # undef linux
 # define LinuxArchitecture
 # ifdef i386
-#  define i386Architecture
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
 #  undef i386
-# endif /* i386 */
+XCOMM Keep cpp from replacing path elements containing i486/i586/i686
+#  ifdef i486
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#   undef i486
+#  endif
+#  ifdef i586
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#   undef i586
+#  endif
+#  ifdef i686
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#   undef i686
+#  endif
+#  ifdef k6
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#   undef k6
+#  endif
+# endif /* k6 */
+# ifdef __i386__
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#  undef __i386__
+# endif /* __i386__ */
+# ifdef __i486__
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#  undef __i486__
+# endif /* __i486__ */
+# ifdef __i586__
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#  undef __i586__
+# endif /* __i586__ */
+# ifdef __i686__
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#  undef __i686__
+# endif /* __i686__ */
+# ifdef __k6__
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#  undef __k6__
+# endif /* __k6__ */
+# ifdef __i386
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#  undef __i386
+# endif /* __i386 */
+# ifdef __i486
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#  undef __i486
+# endif /* __i486 */
+# ifdef __i586
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#  undef __i586
+# endif /* __i586 */
+# ifdef __i686
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#  undef __i686
+# endif /* __i686 */
+# ifdef __k6
+#  ifndef i386Architecture
+#   define i386Architecture
+#  endif
+#  undef __k6
+# endif /* __k6 */
+/* Keep this order! s390x has also __s390__ defined */
+# if defined(__s390x__)
+#  define s390xArchitecture
+#  undef __s390x__
+#  undef __s390__
+# endif
+# ifdef __s390__
+#   define s390Architecture
+#  undef __s390__
+# endif /* s390 */
+# ifdef alpha
+#  define AlphaArchitecture
+#  undef alpha
+# endif /* alpha */
 # ifdef __alpha
 #  define AlphaArchitecture
 #  undef __alpha
 # endif /* __alpha */
+# ifdef __alpha__
+#  ifndef AlphaArchitecture
+#   define AlphaArchitecture
+#  endif
+#  undef __alpha__
+# endif /* __alpha__ */
+# ifdef __arm__
+#  define Arm32Architecture
+#  undef arm
+#  undef __arm
+#  undef __arm__
+# endif
 # ifdef mc68000
 #  define Mc68020Architecture
 #  undef mc68000
 # endif /* mc68000 */
+# ifdef powerpc
+#  define PpcArchitecture
+#  undef powerpc
+# endif
+# ifdef __powerpc__
+#  ifndef PpcArchitecture
+#   define PpcArchitecture
+#  endif
+#  undef __powerpc__
+# endif
+# ifdef sparc
+#  define SparcArchitecture
+#  undef sparc
+# endif
+# ifdef __sparc__
+#  ifndef SparcArchitecture
+#   define SparcArchitecture
+#  endif
+#  undef __sparc__
+# endif
+# ifdef ia64 
+#  define ia64Architecture
+#  undef ia64 
+# endif
+# ifdef __ia64__
+#  ifndef ia64Architecture
+#   define ia64Architecture
+#  endif
+#  undef __ia64__
+# endif
+# if defined(mips) || defined(__mips__)
+#  define MipsArchitecture
+#  undef mips
+#  undef __mips__
+#  if defined(MIPSEL) || defined(_MIPSEL)
+#   define MipselArchitecture
+#  endif
+#  undef MIPSEL
+#  undef _MIPSEL
+# endif
+# ifdef __hppa__
+#  ifndef HPArchitecture
+#   define HPArchitecture
+#  endif
+#  undef __hppa__
+# endif /* __hppa__ */
+# ifdef __sh__
+#  ifndef SuperHArchitecture
+#   define SuperHArchitecture
+#  endif
+#  ifdef __BIG_ENDIAN__
+#   ifndef SuperHebArchitecture
+#    define SuperHebArchitecture
+#   endif
+#  endif
+#  undef __sh__
+# endif /* __sh__ */
+# if defined(__SH3__) || defined(__sh3__)
+#  ifndef SuperH3Architecture
+#   define SuperH3Architecture
+#  endif
+#  undef __SH3__
+#  undef __sh3__
+# endif /* __SH3__ or __sh3__ */
+# ifdef __SH4__
+#  ifdef __SH4_NOFPU__
+#   ifndef SuperH4NOFPUArchitecture
+#    define SuperH4NOFPUArchitecture
+#   endif
+#   undef __SH4_NOFPU__
+#  else
+#   ifndef SuperH4Architecture
+#    define SuperH4Architecture
+#   endif
+#  endif
+#  undef __SH4__
+# endif /* __SH4__ */
+/* for compatibility with 3.3.x */
+# ifdef PpcArchitecture
+#  define PowerPCArchitecture
+# endif
+# if defined(__s390x__)
+#  define s390xArchitecture
+#  undef __s390x__
+# endif
+# if defined(__AMD64__) || defined (__x86_64__)
+#  undef __AMD64__
+#  undef __x86_64__
+#  define AMD64Architecture
+# endif
+# if defined(AMD64__) || defined (x86_64__)
+#  undef AMD64__
+#  undef x86_64__
+#  ifndef AMD64Architecture
+#   define AMD64Architecture
+#  endif
+# endif
 #endif /* linux */
 
 #if (defined(__Lynx__) || defined(Lynx)) && (defined(i386) || defined(__i386__) || defined(__x86__) || defined(__powerpc__) || defined(__sparc__) || defined(sparc))
@@ -652,6 +862,12 @@
 #define i386Architecture
 #endif /* emxos2 */
 
+#ifdef SparcArchitecture
+# if defined(__sparc_v9) || defined(__arch64__)
+#  define Sparc64Architecture
+# endif
+#endif
+
 #ifndef MacroIncludeFile
 XCOMM WARNING:  Imake.cf not configured; guessing at definitions!!!
 XCOMM This might mean that BOOTSTRAPCFLAGS was not set when building imake.
diff -Nur vnc-3.3.7-unixsrc.orig/Xvnc/config/cf/linux.cf vnc-3.3.7-unixsrc/Xvnc/config/cf/linux.cf
--- vnc-3.3.7-unixsrc.orig/Xvnc/config/cf/linux.cf	2002-08-08 13:14:31.000000000 +0200
+++ vnc-3.3.7-unixsrc/Xvnc/config/cf/linux.cf	2004-08-22 11:27:20.308153772 +0200
@@ -40,11 +40,7 @@
 
 /* Set some defaults for the Default* values in case imake doesn't find them */
 #ifndef DefaultLinuxCLibMajorVersion
-#ifdef AlphaArchitecture
 #define DefaultLinuxCLibMajorVersion	6
-#else
-#define DefaultLinuxCLibMajorVersion	5
-#endif
 #endif
 #ifndef DefaultLinuxCLibMinorVersion
 #if DefaultLinuxCLibMajorVersion > 5
@@ -155,69 +151,172 @@
 #endif
 
 #define MkdirHierCmd		mkdir -p
-#if LinuxElfDefault
-#if UseElfFormat
-#define CcCmd			gcc
-#define AsCmd			as
-#define LdCmd			ld
-#define AsmDefines		-D__ELF__
-#define CplusplusCmd		c++
-#if UseStaticTermcapLib
-#define TermcapLibrary		StaticLibrary(/usr/lib/termcap,termcap)
-#endif
-#else
-#ifdef AlphaArchitecture
-#define CcCmd			gcc -b alpha-linuxecoff
-#define CplusplusCmd		g++ -b alpha-linuxecoff
-#define AsCmd			/usr/alpha-linuxecoff/bin/as
-#define LdCmd			/usr/alpha-linuxecoff/bin/ld -m alpha
-#define AsmDefines		-DUSE_GAS -U__ELF__
-#endif /* AlphaArchitecture */
-#ifdef i386Architecture
-#define CcCmd			gcc -b i486-linuxaout
-#define CplusplusCmd		g++ -b i486-linuxaout
-#define AsCmd			/usr/i486-linuxaout/bin/as
-#define LdCmd			/usr/i486-linuxaout/bin/ld -m i386linux
-#define AsmDefines		-DUSE_GAS -U__ELF__
-#endif /* i386Architecture */
-#ifdef Mc68020Architecture
-#define CcCmd			gcc -b m68k-linuxaout
-#define CplusplusCmd		g++ -b m68k-linuxaout
-#define AsCmd			/usr/m68k-linuxaout/bin/as
-#define LdCmd			/usr/m68k-linuxaout/bin/ld -m m68klinux
-#define AsmDefines		-DUSE_GAS -U__ELF__
-#endif /* Mc68020Architecture */
+
+#ifndef HaveLib64
+# if defined (AMD64Architecture) || defined (s390xArchitecture)
+#  define HaveLib64	YES
+# else
+#  define HaveLib64	NO
+# endif
 #endif
+
+#if LinuxElfDefault
+# if UseElfFormat
+#  ifdef MipsArchitecture
+#   ifndef AsCmd
+#     define AsCmd		gcc -c -x assembler-with-cpp
+#   endif
+#   ifndef MakeDllModules
+#     define MakeDllModules	YES
+#   endif
+#  endif /* MipsArchitecure */
+#  if defined (i386Architecture) && ((GccMajorVersion >3) \
+   || ((GccMajorVersion == 3) &&  (GccMinorVersion >= 1)))
+#   ifndef CcCmd
+#    define CcCmd		gcc -m32
+#   endif
+#   ifndef CplusplusCmd
+#    define CplusplusCmd		c++ -m32
+#   endif
+#  endif
+#  ifndef CcCmd
+#   define CcCmd		gcc
+#   define CplusplusCmd		c++
+#  endif
+#  ifndef AsCmd
+#   define AsCmd		CcCmd -c -x assembler
+#  endif
+#  define AsmDefines		-D__ELF__
+#  ifndef LdCmd
+#   define LdCmd		CcCmd -nostdlib
+#  endif
+#  ifndef LinuxUsesNcurses
+#   if LinuxCLibMajorVersion >= 6
+#    define LinuxUsesNcurses	YES
+#   else
+#    define LinuxUsesNcurses	NO
+#   endif
+#  endif
+#  ifndef TermcapLibrary
+#   if UseStaticTermcapLib
+#    if LinuxUsesNcurses
+#     if !HaveLib64
+#      define TermcapLibrary	StaticLibrary(/usr/lib,ncurses)
+#     else
+#      define TermcapLibrary	StaticLibrary(/usr/lib64,ncurses)
+#     endif
+#    else
+#     if !HaveLib64
+#      define TermcapLibrary	StaticLibrary(/usr/lib/termcap,termcap)
+#     else
+#      define TermcapLibrary	StaticLibrary(/usr/lib64/termcap,termcap)
+#     endif
+#    endif
+#   else
+#    if LinuxUsesNcurses
+#     define TermcapLibrary	-lncurses
+#    else
+#     define TermcapLibrary	-ltermcap
+#    endif
+#   endif
+#  endif
+# else
+#  ifdef AlphaArchitecture
+#   define CcCmd		gcc -b alpha-linuxecoff
+#   define CplusplusCmd		g++ -b alpha-linuxecoff
+#   define AsCmd		CcCmd -c -x assembler
+#   define LdCmd		CcCmd -nostdlib -Wl,"-m alpha"
+#   define AsmDefines		-DUSE_GAS -U__ELF__
+#  endif /* AlphaArchitecture */
+#  ifdef HPArchitecture
+#   define CcCmd		gcc
+#   define CplusplusCmd		g++
+#   define AsCmd		CcCmd -c -x assembler
+#   define LdCmd		CcCmd -nostdlib
+#   define AsmDefines		-DUSE_GAS -U__ELF__
+#  endif /* HPArchitecture */
+#  ifdef i386Architecture
+#   define CcCmd		gcc -b i486-linuxaout
+#   define CplusplusCmd		g++ -b i486-linuxaout
+#   define AsCmd		CcCmd -c -x assembler
+#   define LdCmd		CcCmd -nostdlib -Wl,"-m i386linux"
+#   define AsmDefines		-DUSE_GAS -U__ELF__
+#  endif /* i386Architecture */
+#  ifdef ia64Architecture
+#   define CcCmd		gcc
+#   define CplusplusCmd		g++
+#   define AsCmd		CcCmd -c -x assembler
+#   define LdCmd		CcCmd -nostdlib
+#   define AsmDefines		-DUSE_GAS -U__ELF__
+#  endif /* ia64Architecture */
+#  ifdef Mc68020Architecture
+#   define CcCmd		gcc -b m68k-linuxaout
+#   define CplusplusCmd		g++ -b m68k-linuxaout
+#   define AsCmd		CcCmd -c -x assembler
+#   define LdCmd		CcCmd -nostdlib -Wl,"-m m68klinux"
+#   define AsmDefines		-DUSE_GAS -U__ELF__
+#  endif /* Mc68020Architecture */
+#  ifdef AMD64Architecture
+#   define CcCmd		gcc
+#   define CplusplusCmd		g++
+#   define AsCmd		CcCmd -c -x assembler
+#   define LdCmd		CcCmd -nostdlib
+#   define AsmDefines		-DUSE_GAS -U__ELF__
+#  endif /* AMD64Architecture */
+# endif /* UseElfFormat */
 #else
-#if UseElfFormat
-#ifdef AlphaArchitecture
-#define CcCmd			gcc -b alpha-linux
-#define AsCmd			/usr/alpha-linux/bin/as
-#define LdCmd			ld -m elf64alpha
-#define AsmDefines		-D__ELF__
-#define CplusplusCmd		c++ -b alpha-linux
-#endif /* AlphaArchitecture */
-#ifdef i386Architecture
-#define CcCmd			gcc -b i486-linux
-#define AsCmd			/usr/i486-linux/bin/as
-#define LdCmd			ld -m elf_i386
-#define AsmDefines		-D__ELF__
-#define CplusplusCmd		c++ -b i486-linux
-#endif /* i386Architecture */
-#ifdef Mc68020Architecture
-#define CcCmd			gcc -b m68k-linux
-#define AsCmd			/usr/m68k-linux/bin/as
-#define LdCmd			ld -m m68kelf
-#define AsmDefines		-D__ELF__
-#define CplusplusCmd		c++ -b m68k-linux
-#endif /* Mc68020Architecture */
-#else
-#define CcCmd			gcc
-#define AsCmd			as
-#define LdCmd			ld
-#define AsmDefines		-DUSE_GAS
-#endif
+# if UseElfFormat
+#  ifdef AlphaArchitecture
+#   define CcCmd		gcc -b alpha-linux
+#   define AsCmd		CcCmd -c -x assembler
+#   define LdCmd		CcCmd -nostdlib -Wl,"-m elf64alpha"
+#   define AsmDefines		-D__ELF__
+#   define CplusplusCmd		c++ -b alpha-linux
+#  endif /* AlphaArchitecture */
+#  ifdef HPArchitecture
+#   define CcCmd		gcc
+#   define CplusplusCmd		g++
+#   define AsCmd		CcCmd -c -x assembler
+#   define LdCmd		CcCmd -nostdlib
+#   define AsmDefines		-D__ELF__
+#   define SharedLibXau		YES
+#  endif /* HPArchitecture */
+#  ifdef i386Architecture
+#   define CcCmd		gcc -b i486-linux
+#   define AsCmd		CcCmd -c -x assembler
+#   define LdCmd		CcCmd -nosdlib -Wl,"-m elf_i386"
+#   define AsmDefines		-D__ELF__
+#   define CplusplusCmd		c++ -b i486-linux
+#  endif /* i386Architecture */
+#  ifdef ia64Architecture
+#   define CcCmd		gcc
+#   define CplusplusCmd		g++
+#   define AsCmd		CcCmd -c -x assembler
+#   define LdCmd		CcCmd -nostdlib
+#   define AsmDefines		-D__ELF__
+#  endif /* ia64Architecture */
+#  ifdef Mc68020Architecture
+#   define CcCmd		gcc -b m68k-linux
+#   define AsCmd		CcCmd -c -x assembler
+#   define LdCmd		CcCmd -nostdlib -W,l"-m m68kelf"
+#   define AsmDefines		-D__ELF__
+#   define CplusplusCmd		c++ -b m68k-linux
+#  endif /* Mc68020Architecture */
+#  ifdef AMD64Architecture
+#   define CcCmd                 gcc
+#   define CplusplusCmd          g++
+#   define AsCmd                 CcCmd -c -x assembler
+#   define LdCmd                 CcCmd -nostdlib
+#   define AsmDefines            -D__ELF__
+#  endif /* AMD64Architecture */
+# else
+#  define CcCmd			gcc
+#  define AsCmd			CcCmd -c -x assembler
+#  define LdCmd			CcCmd -nostdlib
+#  define AsmDefines		-DUSE_GAS
+# endif
 #endif
+
 #if UseElfFormat
 #ifndef BuildDynamicLoading
 #define BuildDynamicLoading	YES
@@ -246,20 +345,152 @@
 #endif /* i386Architecture */
 
 #ifdef AlphaArchitecture
-#define OptimizedCDebugFlags	-O2
-#define LinuxMachineDefines	-D__alpha__
-#define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME -DPART_NET
-#define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines -D_XSERVER64
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	DefaultGcc2AxpOpt
+# endif
+# define LinuxMachineDefines	-D__alpha__
+# define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines -D_XSERVER64 -DJENSEN_SUPPORT
+# ifdef UseCompaqMathLibrary
+#  define MathLibrary		-lcpml -lm
+# endif
 #endif /* AlphaArchitecture */
 
+#ifdef Arm32Architecture
+# define DefaultCCOptions	-fsigned-char
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	-O3
+# endif
+# define LinuxMachineDefines	-D__arm__ -D__arm32__ -U__arm -Uarm
+# define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
+#endif /* Arm32Achitecture */
+
+#ifdef HPArchitecture
+# define OptimizedCDebugFlags	-O2  GccAliasingArgs
+# define LinuxMachineDefines	-D__hppa__
+# define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
+#endif
+
+#ifdef i386Architecture
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	DefaultGcc2i386Opt
+# endif
+# define LinuxMachineDefines	-D__i386__
+# define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
+#endif /* i386Architecture */
+
+#ifdef ia64Architecture
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	-O2  GccAliasingArgs
+# endif
+# define LinuxMachineDefines	-D__ia64__
+# define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines -D_XSERVER64
+#endif /* ia64Architecture */
+
 #ifdef Mc68020Architecture
-#define OptimizedCDebugFlags	-O2
-#define LinuxMachineDefines 	-D__mc68000__
-#define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME -DPART_NET
-#define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	-O2  GccAliasingArgs
+# endif
+# define LinuxMachineDefines	-D__mc68000__
+# define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
 #endif /* Mc68020Architecture */
 
-#define StandardDefines		-Dlinux LinuxMachineDefines LinuxSourceDefines
+#ifdef MipsArchitecture
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	-O2  GccAliasingArgs
+# endif
+# define LinuxMachineDefines	-D__mips__
+# define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
+#endif
+
+#ifdef PpcArchitecture
+# define DefaultCCOptions	-fsigned-char
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	DefaultGcc2PpcOpt
+# endif
+# define LinuxMachineDefines	-D__powerpc__
+# define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
+#endif /* PpcArchitecture */
+
+#ifdef s390Architecture
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	-O2 -fomit-frame-pointer GccAliasingArgs
+# endif
+# define LinuxMachineDefines	-D__s390__
+# define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
+#endif /* s390Architecture */
+
+#ifdef s390xArchitecture
+/*#define DefaultCCOptions	-fsigned-char */
+#define OptimizedCDebugFlags	-O3 -fomit-frame-pointer
+#define LinuxMachineDefines	-D__s390x__
+#define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME
+#define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines -D_XSERVER64
+#endif /* s390xArchitecture */
+
+#ifdef SparcArchitecture
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	-O2  GccAliasingArgs
+# endif
+# define LinuxMachineDefines	-D__sparc__
+# define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
+# define AsVISOption		-Av9a
+# ifdef Sparc64Architecture
+#  define AsOutputArchSize	64
+# else
+#  define AsOutputArchSize	32
+# endif
+#endif
+
+#ifdef SuperHArchitecture
+# ifndef SuperHArchOptFlags
+#  ifdef SuperH4Architecture
+#   define SuperHArchOptFlags	-m4
+#  elif defined(SuperH4NOFPUArchitecture)
+#   define SuperHArchOptFlags	-m4-nofpu
+#  else
+#   define SuperHArchOptFlags	-m3
+#  endif
+# endif 
+# ifndef SuperHebArchitecture
+#  ifdef SuperHebArchitecture
+#   define SuperHEndianFlags	-mb
+#  else
+#   define SuperHEndianFlags	-ml
+#  endif
+# endif
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	-O2 SuperHArchOptFlags SuperHEndianFlags  GccAliasingArgs
+# endif
+# define LinuxMachineDefines	-D__sh__
+# define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
+# ifndef DoLoadableServer
+#  define DoLoadableServer NO
+# endif
+#endif
+
+#ifdef AMD64Architecture
+# ifndef OptimizedCDebugFlags
+#  define OptimizedCDebugFlags	DefaultGcc2AMD64Opt
+# endif
+# define LinuxMachineDefines   -D__AMD64__
+# define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines -D_XSERVER64
+#endif /* AMD64Architecture */
+
+#ifndef StandardDefines
+# define StandardDefines	-Dlinux LinuxMachineDefines LinuxSourceDefines
+#endif
 
 #if UseElfFormat
 #define DlLibrary		-rdynamic -ldl
@@ -303,5 +534,17 @@
 #define MakeNamedTargetSubdir(dir,flags,subname)\
 	$(MAKE) -C dir $(MFLAGS) $(PARALLELMFLAGS) flags subname
 
+#if HaveLib64
+# ifndef LibDirName
+#  define LibDirName		lib64
+# endif
+# ifndef SystemUsrLibDir
+#  define SystemUsrLibDir	/usr/lib64
+# endif
+# ifndef TkLibDir
+#  define TkLibDir		/usr/lib64
+# endif
+#endif
+
 #include <lnxLib.rules>
 #include <xfree86.cf>
diff -Nur vnc-3.3.7-unixsrc.orig/Xvnc/config/cf/vnc.def vnc-3.3.7-unixsrc/Xvnc/config/cf/vnc.def
--- vnc-3.3.7-unixsrc.orig/Xvnc/config/cf/vnc.def	2003-02-07 13:41:05.000000000 +0100
+++ vnc-3.3.7-unixsrc/Xvnc/config/cf/vnc.def	2004-08-22 11:28:06.252072009 +0200
@@ -117,7 +117,7 @@
 #ifdef SunArchitecture
 #define ProjectRoot /usr/openwin
 #else
-#ifdef i386Architecture
+#ifdef LinuxArchitecture
 #define ProjectRoot /usr/X11R6
 #endif
 #endif
diff -Nur vnc-3.3.7-unixsrc.orig/Xvnc/config/cf/xfree86.cf vnc-3.3.7-unixsrc/Xvnc/config/cf/xfree86.cf
--- vnc-3.3.7-unixsrc.orig/Xvnc/config/cf/xfree86.cf	2002-05-11 12:57:05.000000000 +0200
+++ vnc-3.3.7-unixsrc/Xvnc/config/cf/xfree86.cf	2004-08-22 12:51:14.836262700 +0200
@@ -1007,12 +1007,41 @@
 #endif
 #endif
 
+#if  (GccMajorVersion == 3) &&  (GccMinorVersion >= 1)
+# define GccAliasingArgs      -fno-strict-aliasing
+#else
+# define GccAliasingArgs      /* */
+#endif
+
 #if HasGcc2 && defined(i386Architecture)
-#ifndef DefaultGcc2i386Opt
-#define DefaultGcc2i386Opt -O2 -fno-strength-reduce
+# ifndef DefaultGcc2i386Opt
+#  define DefaultGcc2i386Opt -O2 -fno-strength-reduce GccAliasingArgs
+# endif
+#endif
+
+#if HasGcc2 && defined(AMD64Architecture)
+# ifndef DefaultGcc2AMD64Opt
+#  define DefaultGcc2AMD64Opt -O2 -fno-strength-reduce GccAliasingArgs
+# endif
+#endif
+
+#if HasGcc2 && defined(AlphaArchitecture)
+# ifndef DefaultGcc2AxpOpt
+#  define DefaultGcc2AxpOpt -O2 GccAliasingArgs
+# endif
 #endif
+
+#if HasGcc2 && defined(PpcArchitecture)
+# ifndef DefaultGcc2PpcOpt
+#  define DefaultGcc2PpcOpt -O2 GccAliasingArgs
+# endif
 #endif
 
+#ifndef DefaultGcc2DebugOpt
+# define DefaultGcc2DebugOpt
+#endif
+
+
 #ifndef DontDefineXF86Contrib
 #define XF86Contrib
 #endif
diff -Nur vnc-3.3.7-unixsrc.orig/Xvnc/config/imake/imakemdep.h vnc-3.3.7-unixsrc/Xvnc/config/imake/imakemdep.h
--- vnc-3.3.7-unixsrc.orig/Xvnc/config/imake/imakemdep.h	2002-05-11 12:57:05.000000000 +0200
+++ vnc-3.3.7-unixsrc/Xvnc/config/imake/imakemdep.h	2004-08-22 11:53:00.479173080 +0200
@@ -294,9 +294,68 @@
 #ifdef unix
 	"-Uunix",	/* remove unix symbol so that filename unix.c okay */
 #endif
-#if defined(__386BSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__FreeBSD__) || defined(MACH)
+#if defined(__386BSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || \
+    defined(__FreeBSD__) || defined(MACH) || defined(linux) || \
+    defined(__GNU__) || defined(__bsdi__) || defined(__GNUC__)
 # ifdef __i386__
 	"-D__i386__",
+#  if defined(__GNUC__) && (__GNUC__ >= 3)
+	"-m32",
+#  endif
+# endif
+# ifdef __i486__
+	"-D__i486__",
+# endif
+# ifdef __i586__
+	"-D__i586__",
+# endif
+# ifdef __i686__
+	"-D__i686__",
+# endif
+# ifdef __k6__
+	"-D__k6__",
+# endif
+# ifdef __ia64__
+	"-D__ia64__",
+# endif
+# ifdef __amd64__
+	"-D__amd64__",
+# endif
+# ifdef __x86_64__
+	"-D__amd64__",
+# endif
+# ifdef __s390__
+	"-D__s390__",
+# endif
+# ifdef __alpha__
+	"-D__alpha__",
+# endif
+# ifdef __arm__
+	"-D__arm__",
+# endif
+# ifdef __s390x__
+       "-D__s390x__",
+# endif
+# ifdef __sparc__
+	"-D__sparc__",
+# endif
+# ifdef __m68k__
+	"-D__m68k__",
+# endif
+# ifdef __sh__
+	"-D__sh__",
+# endif
+# ifdef __sh3__
+	"-D__sh3__",
+# endif
+# ifdef __SH3__
+	"-D__SH3__",
+# endif
+# ifdef __SH4__
+	"-D__SH4__",
+# endif
+# ifdef __SH4NOFPU__
+	"-D__SH4_NOFPU__",
 # endif
 # ifdef __GNUC__
 	"-traditional",
@@ -489,7 +548,6 @@
 	"-DNCR",	/* NCR */
 #endif
 #ifdef linux
-        "-traditional",
         "-Dlinux",
 #endif
 #ifdef __uxp__
@@ -715,9 +773,18 @@
 #ifdef sparc
 	{"sparc", "1"},
 #endif
+#ifdef __sparc
+	{"__sparc", "1"},
+#endif
+#ifdef __sparcv9
+	{"__sparcv9", "1"},
+#endif
 #ifdef __sparc__
 	{"__sparc__", "1"},
 #endif
+#ifdef __sparcv9__
+	{"__sparcv9__", "1"},
+#endif
 #ifdef hpux
 	{"hpux", "1"},
 #endif
@@ -805,6 +872,12 @@
 #ifdef m68k
         {"m68k", "1"},
 #endif
+#ifdef M68k
+        {"M68k", "1"},
+#endif
+#ifdef __m68k__
+	{"__m68k__", "1"},
+#endif
 #ifdef m88k
         {"m88k", "1"},
 #endif
@@ -832,6 +905,9 @@
 #ifdef __osf__
 	{"__osf__", "1"},
 #endif
+#ifdef __amiga__
+	{"__amiga__", "1"},
+#endif
 #ifdef __alpha
 	{"__alpha", "1"},
 #endif
@@ -890,6 +966,24 @@
 #ifdef __sgi
 	{"__sgi", "1"},
 #endif
+#ifdef _MIPS_FPSET
+	{"_MIPS_FPSET", DEF_STRINGIFY(_MIPS_FPSET)},
+#endif
+#ifdef _MIPS_ISA
+	{"_MIPS_ISA", DEF_STRINGIFY(_MIPS_ISA)},
+#endif
+#ifdef _MIPS_SIM
+	{"_MIPS_SIM", DEF_STRINGIFY(_MIPS_SIM)},
+#endif
+#ifdef _MIPS_SZINT
+	{"_MIPS_SZINT", DEF_STRINGIFY(_MIPS_SZINT)},
+#endif
+#ifdef _MIPS_SZLONG
+	{"_MIPS_SZLONG", DEF_STRINGIFY(_MIPS_SZLONG)},
+#endif
+#ifdef _MIPS_SZPTR
+	{"_MIPS_SZPTR", DEF_STRINGIFY(_MIPS_SZPTR)},
+#endif
 #ifdef __FreeBSD__
 	{"__FreeBSD__", "1"},
 #endif
@@ -899,9 +993,107 @@
 #ifdef __NetBSD__
 	{"__NetBSD__", "1"},
 #endif
+#ifdef __GNU__
+	{"__GNU__", "1"},
+#endif
 #ifdef __ELF__
 	{"__ELF__", "1"},
 #endif
+#ifdef __UNIXOS2__
+	{"__UNIXOS2__", "1"},
+#endif
+#if defined(__QNX__)
+        {"__QNX__", "1"},
+#endif
+#ifdef __QNXNTO__
+        {"__QNXNTO__", "1"},
+#endif
+# ifdef __powerpc__
+	{"__powerpc__", "1"},
+# endif
+# ifdef PowerMAX_OS
+	{"PowerMAX_OS", "1"},
+# endif
+# ifdef ia64
+	{"ia64", "1"},
+# endif
+# ifdef __ia64__
+	{"__ia64__", "1"},
+# endif
+# if defined (amd64) || defined (x86_64)
+	{"amd64", "1"},
+	{"x86_64", "1"},
+# endif
+# if defined (__amd64__) || defined (__x86_64__)
+	{"__amd64__", "1"},
+	{"__x86_64__", "1"},
+# endif
+# ifdef __i386
+	{"__i386", "1"},
+# endif
+# ifdef __i386__
+	{"__i386__", "1"},
+# endif
+# ifdef __i486__
+	{"__i486__", "1"},
+# endif
+# ifdef __i586__
+	{"__i586__", "1"},
+# endif
+# ifdef __i686__
+	{"__i686__", "1"},
+# endif
+# ifdef __k6__
+	{"__k6__", "1"},
+# endif
+# ifdef i386
+	{"i386", "1"},
+# endif
+# ifdef i486
+	{"i486", "1"},
+# endif
+# ifdef i586
+	{"i586", "1"},
+# endif
+# ifdef i686
+	{ "i686", "1"},
+# endif
+# ifdef k6
+	{"k6", "1"},
+# endif
+# ifdef sparc
+	{"sparc", "1"},
+# endif
+# ifdef __sparc__
+	{"__sparc__", "1"},
+# endif
+# ifdef __s390__
+	{"__s390__", "1"},
+# endif
+# ifdef __sh__
+	{"__sh__", "1"},
+# endif
+# ifdef __sh3_
+	{"__sh3__", "1"},
+# endif
+# ifdef __SH3__
+	{"__SH3__", "1"},
+# endif
+# ifdef __SH4__
+	{"__SH4__", "1"},
+# endif
+# ifdef __SH4NOFPU__
+	{"__SH4NOFPU__", "1"},
+# endif
+#if defined(__ppc__)
+        {"__ppc__", "1"},
+#endif
+#if defined(__BIG_ENDIAN__)
+      {"__BIG_ENDIAN__", "1"},
+#endif
+#if defined(__LITTLE_ENDIAN__)
+      {"__LITTLE_ENDIAN__", "1"},
+#endif
 #ifdef __EMX__
 	{"__EMX__", "1"},
 #endif
diff -Nur vnc-3.3.7-unixsrc.orig/Xvnc/programs/Xserver/cfb/Imakefile vnc-3.3.7-unixsrc/Xvnc/programs/Xserver/cfb/Imakefile
--- vnc-3.3.7-unixsrc.orig/Xvnc/programs/Xserver/cfb/Imakefile	2002-05-11 12:57:06.000000000 +0200
+++ vnc-3.3.7-unixsrc/Xvnc/programs/Xserver/cfb/Imakefile	2004-08-22 11:30:08.180667941 +0200
@@ -205,10 +205,10 @@
 	$(RM) stipsparcte.s
 
 stipsparc.o: stipsparc.s
-	$(CPP) stipsparc.s | $(AS) -o $@ -
+	$(CPP) AsmDefines stipsparc.s | $(AS) -o $@ -
 
 stipsparcte.o: stipsparcte.s
-	$(CPP) -DTETEXT stipsparcte.s | $(AS) -o $@ -
+	$(CPP) AsmDefines -DTETEXT stipsparcte.s | $(AS) -o $@ -
 #endif
 #endif
 
@@ -221,10 +221,10 @@
 	$(RM) stipsprcte32.s
 
 stipsprc32.o: stipsprc32.s
-	$(CPP) stipsprc32.s | $(AS) -o $@ -
+	$(CPP) AsmDefines stipsprc32.s | $(AS) -o $@ -
 
 stipsprcte32.o: stipsprcte32.s
-	$(CPP) -DTETEXT stipsprcte32.s | $(AS) -o $@ -
+	$(CPP) AsmDefines -DTETEXT stipsprcte32.s | $(AS) -o $@ -
 #endif
 #endif
 
diff -Nur vnc-3.3.7-unixsrc.orig/Xvnc/programs/Xserver/cfb/stipsparc.s vnc-3.3.7-unixsrc/Xvnc/programs/Xserver/cfb/stipsparc.s
--- vnc-3.3.7-unixsrc.orig/Xvnc/programs/Xserver/cfb/stipsparc.s	2002-05-11 12:57:06.000000000 +0200
+++ vnc-3.3.7-unixsrc/Xvnc/programs/Xserver/cfb/stipsparc.s	2004-08-22 11:32:01.956236729 +0200
@@ -85,7 +85,7 @@
 #define ForEachBits	LY4
 #define NextBits	LY5
 
-#ifdef SVR4
+#if defined(SVR4) || (defined(linux) && defined(__ELF__))
 #ifdef TETEXT
 #define	_cfbStippleStack	cfbStippleStackTE
 #else
diff -Nur vnc-3.3.7-unixsrc.orig/Xvnc/programs/Xserver/cfb/stipsprc32.s vnc-3.3.7-unixsrc/Xvnc/programs/Xserver/cfb/stipsprc32.s
--- vnc-3.3.7-unixsrc.orig/Xvnc/programs/Xserver/cfb/stipsprc32.s	2002-05-11 12:57:06.000000000 +0200
+++ vnc-3.3.7-unixsrc/Xvnc/programs/Xserver/cfb/stipsprc32.s	2004-08-22 11:32:18.926145824 +0200
@@ -81,7 +81,7 @@
 #define ForEachBits	LY4
 #define NextBits	LY5
 
-#ifdef SVR4
+#if defined(SVR4) || (defined(linux) && defined(__ELF__))
 #ifdef TETEXT
 #define	_cfb32StippleStack	cfb32StippleStackTE
 #else
diff -Nur vnc-3.3.7-unixsrc.orig/Xvnc/programs/Xserver/include/servermd.h vnc-3.3.7-unixsrc/Xvnc/programs/Xserver/include/servermd.h
--- vnc-3.3.7-unixsrc.orig/Xvnc/programs/Xserver/include/servermd.h	2002-05-11 12:57:06.000000000 +0200
+++ vnc-3.3.7-unixsrc/Xvnc/programs/Xserver/include/servermd.h	2004-08-22 12:43:57.041690141 +0200
@@ -132,7 +132,42 @@
 
 #endif /* vax */
 
-#if (defined(Lynx) && defined(__powerpc__))
+#ifdef __arm32__
+
+#define IMAGE_BYTE_ORDER        LSBFirst
+
+# if defined(XF86MONOVGA) || defined(XF86VGA16) || defined(XF86MONO)
+#  define BITMAP_BIT_ORDER      MSBFirst
+# else
+#  define BITMAP_BIT_ORDER      LSBFirst
+# endif
+
+# if defined(XF86MONOVGA) || defined(XF86VGA16)
+#  define BITMAP_SCANLINE_UNIT  8
+# endif
+
+#define GLYPHPADBYTES           4
+#define GETLEFTBITS_ALIGNMENT   1
+#define LARGE_INSTRUCTION_CACHE
+#define AVOID_MEMORY_READ
+
+#endif /* __arm32__ */
+
+#if defined (hpux)
+
+#define IMAGE_BYTE_ORDER	MSBFirst
+#define BITMAP_BIT_ORDER	MSBFirst
+#define GLYPHPADBYTES		4	/* to make fb work */
+#define GETLEFTBITS_ALIGNMENT	1	/* PA forces longs to 4 */
+					/* byte boundries */
+#define AVOID_MEMORY_READ
+#define FAST_CONSTANT_OFFSET_MODE
+#define LARGE_INSTRUCTION_CACHE
+#define PLENTIFUL_REGISTERS
+
+#endif /* hpux */
+
+#if defined(__powerpc__)
 
 /* For now this is for Xvfb only */
 #define IMAGE_BYTE_ORDER        MSBFirst
@@ -147,14 +182,33 @@
 
 #define FAST_MEMCPY
 
-#endif /* LynxOS PowerPC */
+#endif /* PowerPC */
+
+#if defined(__sh__)
+
+#if defined(__BIG_ENDIAN__)
+# define IMAGE_BYTE_ORDER	MSBFirst
+# define BITMAP_BIT_ORDER	MSBFirst
+# define GLYPHPADBYTES		4
+# define GETLEFTBITS_ALIGNMENT	1
+#else
+# define IMAGE_BYTE_ORDER	LSBFirst
+# define BITMAP_BIT_ORDER	LSBFirst
+# define GLYPHPADBYTES		4
+# define GETLEFTBITS_ALIGNMENT	1
+#endif
+
+#define AVOID_MEMORY_READ
+#define FAST_CONSTANT_OFFSET_MODE
+#define LARGE_INSTRUCTION_CACHE
+#define PLENTIFUL_REGISTERS
+
+#endif /* SuperH */
+
 
 #if (defined(sun) && !(defined(i386) && defined(SVR4))) || \
-    (defined(AMOEBA) && (defined(sparc) || defined(mc68000))) || \
     (defined(__uxp__) && (defined(sparc) || defined(mc68000))) || \
-    (defined(Lynx) && defined(__sparc__)) || \
-    ((defined(__NetBSD__) || defined(__OpenBSD__)) && \
-     (defined(__sparc__) || defined(__mc68000__)))
+    defined(__sparc__) || defined(__mc68000__)
 
 #if defined(sun386) || defined(sun5)
 # define IMAGE_BYTE_ORDER	LSBFirst        /* Values for the SUN only */
@@ -214,20 +268,6 @@
 
 #endif /* ibm */
 
-#ifdef hpux
-
-#define IMAGE_BYTE_ORDER	MSBFirst        /* Values for the HP only */
-#define BITMAP_BIT_ORDER	MSBFirst
-#define	GLYPHPADBYTES		2		/* to match product server */
-#define	GETLEFTBITS_ALIGNMENT	4		/* PA forces longs to 4 */
-						/* byte boundries */
-#define AVOID_MEMORY_READ
-#define FAST_CONSTANT_OFFSET_MODE
-#define LARGE_INSTRUCTION_CACHE
-#define PLENTIFUL_REGISTERS
-
-#endif /* hpux */
-
 #if defined (M4310) || defined(M4315) || defined(M4317) || defined(M4319) || defined(M4330)
 
 #define IMAGE_BYTE_ORDER	MSBFirst        /* Values for Pegasus only */
@@ -306,6 +346,80 @@
 
 #endif /* alpha */
 
+#if defined (linux) && defined (__s390__)
+
+#define IMAGE_BYTE_ORDER      	MSBFirst
+#define BITMAP_BIT_ORDER      	MSBFirst
+#define GLYPHPADBYTES         	4
+#define GETLEFTBITS_ALIGNMENT  1	
+
+#define BITMAP_SCANLINE_UNIT	8
+#define LARGE_INSTRUCTION_CACHE
+#define FAST_CONSTANT_OFFSET_MODE
+#define FAST_UNALIGNED_READ
+
+#define FAST_MEMCPY
+
+#endif /* linux/s390 */
+
+#if defined (linux) && defined (__s390x__)
+
+#define IMAGE_BYTE_ORDER       MSBFirst
+#define BITMAP_BIT_ORDER       MSBFirst
+#define GLYPHPADBYTES          4
+#define GETLEFTBITS_ALIGNMENT  1
+
+#define BITMAP_SCANLINE_UNIT	8
+#define LARGE_INSTRUCTION_CACHE
+#define FAST_CONSTANT_OFFSET_MODE
+#define FAST_UNALIGNED_READ
+
+#define FAST_MEMCPY
+#endif /* linux/s390x */
+
+
+#if defined(__ia64__) || defined(ia64)
+# define IMAGE_BYTE_ORDER	LSBFirst
+
+# if defined(XF86MONOVGA) || defined(XF86VGA16) || defined(XF86MONO)
+#  define BITMAP_BIT_ORDER      MSBFirst
+# else
+#  define BITMAP_BIT_ORDER      LSBFirst
+# endif
+
+# if defined(XF86MONOVGA) || defined(XF86VGA16)
+#  define BITMAP_SCANLINE_UNIT  8
+# endif
+
+# define GLYPHPADBYTES		4
+# define GETLEFTBITS_ALIGNMENT	1
+# define FAST_CONSTANT_OFFSET_MODE
+# define LARGE_INSTRUCTION_CACHE
+# define PLENTIFUL_REGISTERS
+
+#endif /* ia64 */
+
+#if defined(__amd64__) || defined(amd64)
+# define IMAGE_BYTE_ORDER	LSBFirst
+
+# if defined(XF86MONOVGA) || defined(XF86VGA16) || defined(XF86MONO)
+#  define BITMAP_BIT_ORDER      MSBFirst
+# else
+#  define BITMAP_BIT_ORDER      LSBFirst
+# endif
+
+# if defined(XF86MONOVGA) || defined(XF86VGA16)
+#  define BITMAP_SCANLINE_UNIT  8
+# endif
+
+# define GLYPHPADBYTES		4
+# define GETLEFTBITS_ALIGNMENT	1
+# define LARGE_INSTRUCTION_CACHE
+# define FAST_CONSTANT_OFFSET_MODE
+/* ???? */
+# define FAST_UNALIGNED_READS
+#endif /* AMD64 */
+
 #ifdef stellar
 
 #define IMAGE_BYTE_ORDER	MSBFirst       /* Values for the stellar only*/
@@ -336,8 +450,14 @@
 
 #endif /* luna */
 
-#if (defined(i386) && (defined(SVR4) || defined(SYSV) || (defined(sun) && defined(SVR4))) || defined(__bsdi__) || (defined(__NetBSD__) && defined(__i386__)) || (defined(__OpenBSD__) && defined(__i386__)) || defined(__FreeBSD__) || defined(MACH386) || (defined(linux) && !defined(__mc68000__)) || (defined(AMOEBA) && defined(i80386)) || defined(MINIX) || defined(__EMX__) || (defined(Lynx) && defined(__i386__)))
-
+#if	(defined(SVR4) && defined(i386)) || \
+	defined(__alpha__) || defined(__alpha) || \
+	defined(__i386__) || \
+	defined(__UNIXOS2__) || \
+	defined(__OS2ELF__) || \
+	defined(__QNX__) || \
+	defined(__s390x__) || defined(__s390__)
+  
 #ifndef IMAGE_BYTE_ORDER
 #define IMAGE_BYTE_ORDER	LSBFirst
 #endif
@@ -420,6 +540,14 @@
 
 #endif
 
+/* linux on the Compaq Itsy */
+#if defined(linux) && defined(__arm__)
+#define IMAGE_BYTE_ORDER	LSBFirst
+#define BITMAP_BIT_ORDER	LSBFirst
+#define GLYPHPADBYTES		4
+#define GETLEFTBITS_ALIGNMENT	1
+#endif
+ 
 /* size of buffer to use with GetImage, measured in bytes. There's obviously
  * a trade-off between the amount of stack (or whatever ALLOCATE_LOCAL gives
  * you) used and the number of times the ddx routine has to be called.
