--- vnc-4_1_2-unixsrc/unix/xorg-server/hw/xfree86/common/compiler.h.ppc64	2006-11-14 16:10:58.000000000 +0100
+++ vnc-4_1_2-unixsrc/unix/xorg-server/hw/xfree86/common/compiler.h	2006-11-14 16:14:11.000000000 +0100
@@ -1081,7 +1081,7 @@
 #if defined(linux) && defined(__powerpc64__)
 # include <linux/version.h>
 # if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 0)
-#  include <asm/memory.h>
+/*#  include <asm/memory.h>*/
 # endif
 #endif /* defined(linux) && defined(__powerpc64__) */
 #ifndef eieio /* We deal with arch-specific eieio() routines above... */
