diff -uNr vnc-3.3.2r2.orig/Xvnc/config/cf/Imake.cf vnc-3.3.2r2/Xvnc/config/cf/Imake.cf
--- vnc-3.3.2r2.orig/Xvnc/config/cf/Imake.cf	Sun Jul 27 11:41:02 1997
+++ vnc-3.3.2r2/Xvnc/config/cf/Imake.cf	Tue Nov  3 15:35:01 1998
@@ -540,8 +540,13 @@
 #endif /* WIN32 */
 
 #ifdef linux 
-# define MacroIncludeFile <linux.cf>
-# define MacroFile linux.cf
+# ifdef PPC
+#  define MacroIncludeFile <linux-pmac.cf>
+#  define MacroFile       linux-pmac.cf
+# else
+#  define MacroIncludeFile <linux.cf>
+#  define MacroFile linux.cf
+# endif
 # undef linux
 # define LinuxArchitecture
 # ifdef i386
@@ -556,6 +561,9 @@
 #  define Mc68020Architecture
 #  undef mc68000
 # endif /* mc68000 */
+# ifdef PPC
+#  define PPCArchitecture
+# endif
 #endif /* linux */
 
 #if (defined(__Lynx__) || defined(Lynx)) && (defined(i386) || defined(__i386__) || defined(__x86__) || defined(__powerpc__) || defined(__sparc__) || defined(sparc))
diff -uNr vnc-3.3.2r2.orig/Xvnc/config/cf/Imakefile vnc-3.3.2r2/Xvnc/config/cf/Imakefile
--- vnc-3.3.2r2.orig/Xvnc/config/cf/Imakefile	Sun Jul 27 11:41:03 1997
+++ vnc-3.3.2r2/Xvnc/config/cf/Imakefile	Tue Nov  3 15:35:01 1998
@@ -90,6 +90,7 @@
 ibm.cf \
 isc.cf \
 linux.cf \
+linux-pmac.cf \
 luna.cf \
 macII.cf \
 mach.cf \
diff -uNr vnc-3.3.2r2.orig/Xvnc/config/cf/linux-pmac.cf vnc-3.3.2r2/Xvnc/config/cf/linux-pmac.cf
--- vnc-3.3.2r2.orig/Xvnc/config/cf/linux-pmac.cf	Thu Jan  1 09:00:00 1970
+++ vnc-3.3.2r2/Xvnc/config/cf/linux-pmac.cf	Tue Nov  3 15:35:01 1998
@@ -0,0 +1,95 @@
+
+#ifndef OSName
+#define OSName			Linux
+#endif
+#ifndef OSVendor
+#define OSVendor		Linux PPC
+#endif
+#ifndef OSMajorVersion
+#define OSMajorVersion		2
+#endif
+#ifndef OSMinorVersion
+#define OSMinorVersion		0
+#endif
+#ifndef OSTeenyVersion
+#define OSTeenyVersion		33
+#endif
+
+/* #define HasNdbm			YES */
+/* #define DBMLibrary		-lgdbm -lz */
+#define UseElfFormat		YES
+#define HasGcc			YES
+#define HasGcc2			YES
+/* Create static and shared libraries */
+#define HasSharedLibraries	YES
+#define ForceNormalLib		YES
+#define HasGcc2ForCplusplus	YES
+#define GccUsesGas		YES
+#define UseGas			YES
+#define GnuCpp			YES
+#define HasShadowPasswd		YES
+#define HasLibCrypt		YES
+#define HasPutenv		YES
+#define HasShm			YES
+#define HasSockets		YES
+#define HasZlib                 YES 
+
+#define AvoidNullMakeCommand	YES
+#define StripInstalledPrograms	YES
+#define CompressAllFonts	YES
+#define Malloc0ReturnsNull	YES
+#define NeedConstPrototypes	YES
+#define NeedFunctionPrototypes	YES
+#define NeedNestedPrototypes	YES
+#define NeedVarargsPrototypes	YES
+#define NeedWidePrototypes	NO
+#define SetTtyGroup		YES
+
+#define CcCmd			gcc
+#define DefaultCCOptions	-fsigned-char
+#define AsCmd			as
+#define LdCmd			ld
+#define AsmDefines		-DUSE_GAS -U__ELF__
+#define MkdirHierCmd		mkdir -p
+#define CppCmd			/lib/cpp
+#define YaccCmd			byacc
+#define LexCmd			flex -l
+#define LexLib			-lfl
+#define PreProcessCmd		CcCmd -E
+#define PostIncDir		`CcCmd --print-libgcc-file-name | sed 's/libgcc.a/include/'`
+#define LdCombineFlags		-r
+#define OptimizedCDebugFlags	-O2 
+#define StandardDefines		-Dlinux -DPPC -D__powerpc__ -D_POSIX_SOURCE \
+				-D_BSD_SOURCE -D_SVID_SOURCE -DX_LOCALE
+#define XawI18nDefines		-DUSE_XWCHAR_STRING -DUSE_XMBTOWC
+#define HasWChar32		YES
+#define StandardCppDefines	-traditional StandardDefines
+#define ServerOSDefines		-DDDXTIME
+#define ServerExtraDefines	-DGCCUSESGAS
+#define ConnectionFlags		-DUNIXCONN -DTCPCONN
+#define InstallXserverSetUID	YES
+#define InstUidFlags		-s -m 4711
+
+#define	ServerExtraSysLibs	
+
+/* Some of these man page defaults are overriden in the above OS sections */
+#ifndef ManSuffix
+# define ManSuffix	1x
+#endif
+#ifndef ManDir
+# define ManDir		$(MANSOURCEPATH)1
+#endif
+#ifndef LibManSuffix
+# define LibManSuffix	3x
+#endif
+#ifndef LibmanDir
+# define LibmanDir	$(MANSOURCEPATH)3
+#endif
+#ifndef FileManSuffix
+# define FileManSuffix	5x
+#endif
+#ifndef FileManDir
+# define FileManDir	$(MANSOURCEPATH)5
+#endif
+
+#include <lnxLib.rules>
diff -uNr vnc-3.3.2r2.orig/Xvnc/config/cf/vnc.def vnc-3.3.2r2/Xvnc/config/cf/vnc.def
--- vnc-3.3.2r2.orig/Xvnc/config/cf/vnc.def	Thu Jul 16 00:44:55 1998
+++ vnc-3.3.2r2/Xvnc/config/cf/vnc.def	Tue Nov  3 15:35:01 1998
@@ -117,7 +117,7 @@
 #ifdef SunArchitecture
 #define ProjectRoot /usr/openwin
 #else
-#ifdef i386Architecture
+#ifdef i386Architecture || PPCArchitecture
 #define ProjectRoot /usr/X11R6
 #endif
 #endif
diff -uNr vnc-3.3.2r2.orig/Xvnc/config/imake/imakemdep.h vnc-3.3.2r2/Xvnc/config/imake/imakemdep.h
--- vnc-3.3.2r2.orig/Xvnc/config/imake/imakemdep.h	Sun Jul 27 11:41:05 1997
+++ vnc-3.3.2r2/Xvnc/config/imake/imakemdep.h	Tue Nov  3 18:20:00 1998
@@ -766,6 +766,12 @@
 #ifdef mc68020
 	{"mc68020", "1"},
 #endif
+#ifdef PPC
+	{"PPC", "1"},
+#endif
+#ifdef __powerpc__
+	{"__powerpc__", "1"},
+#endif
 #ifdef __GNUC__
 	{"__GNUC__", "1"},
 #endif
diff -uNr vnc-3.3.2r2.orig/Xvnc/include/Xos.h vnc-3.3.2r2/Xvnc/include/Xos.h
--- vnc-3.3.2r2.orig/Xvnc/include/Xos.h	Mon Mar  2 20:56:46 1998
+++ vnc-3.3.2r2/Xvnc/include/Xos.h	Tue Nov  3 15:35:01 1998
@@ -118,7 +118,11 @@
  */
 #if (defined(X_NOT_STDC_ENV) || (defined(sun) && !defined(SVR4)) || defined(macII)) && !defined(__GLIBC__)
 #ifndef strerror
+#if defined(linux) && defined(PPC)
+extern const char *const sys_errlist[];
+#else
 extern char *sys_errlist[];
+#endif
 extern int sys_nerr;
 #define strerror(n) \
     (((n) >= 0 && (n) < sys_nerr) ? sys_errlist[n] : "unknown error")
diff -uNr vnc-3.3.2r2.orig/Xvnc/programs/Xserver/include/servermd.h vnc-3.3.2r2/Xvnc/programs/Xserver/include/servermd.h
--- vnc-3.3.2r2.orig/Xvnc/programs/Xserver/include/servermd.h	Mon Jul 28 23:17:34 1997
+++ vnc-3.3.2r2/Xvnc/programs/Xserver/include/servermd.h	Tue Nov  3 17:33:02 1998
@@ -336,7 +336,18 @@
 
 #endif /* luna */
 
-#if (defined(i386) && (defined(SVR4) || defined(SYSV) || (defined(sun) && defined(SVR4))) || defined(__bsdi__) || (defined(__NetBSD__) && defined(__i386__)) || (defined(__OpenBSD__) && defined(__i386__)) || defined(__FreeBSD__) || defined(MACH386) || (defined(linux) && !defined(__mc68000__)) || (defined(AMOEBA) && defined(i80386)) || defined(MINIX) || defined(__EMX__) || (defined(Lynx) && defined(__i386__)))
+#if (defined(i386) && (defined(SVR4) || defined(SYSV) || \
+      (defined(sun) && defined(SVR4))) || \
+    defined(__bsdi__) || \
+    (defined(__NetBSD__) && defined(__i386__)) || \
+    (defined(__OpenBSD__) && defined(__i386__)) || \
+    defined(__FreeBSD__) || \
+    defined(MACH386) || \
+    (defined(linux) && ! (defined(__mc68000__) || defined(PPC))) || \
+    (defined(AMOEBA) && defined(i80386)) || \
+    defined(MINIX) || \
+    defined(__EMX__) || \
+    (defined(Lynx) && defined(__i386__)))
 
 #ifndef IMAGE_BYTE_ORDER
 #define IMAGE_BYTE_ORDER	LSBFirst
@@ -380,6 +391,18 @@
 #define GETLEFTBITS_ALIGNMENT  1
 
 #endif /* linux/m68k */
+
+#if defined (linux) && defined (PPC)
+
+#define IMAGE_BYTE_ORDER       MSBFirst
+#define BITMAP_BIT_ORDER       MSBFirst
+#define GLYPHPADBYTES          4
+#define GETLEFTBITS_ALIGNMENT  1
+
+#define LARGE_INSTRUCTION_CACHE
+#define FAST_CONSTANT_OFFSET_MODE
+
+#endif /* linux/PPC */
 
 #ifdef sgi
 
--- vnc_unixsrc/Xvnc/programs/Xserver/os/WaitFor.c~	Mon Mar  2 12:56:36 1998
+++ vnc_unixsrc/Xvnc/programs/Xserver/os/WaitFor.c	Mon Jul  1 01:04:09 2002
@@ -57,6 +57,9 @@
  *
  *****************************************************************/
 
+
+#define _GNU_SOURCE
+
 #ifdef WIN32
 #include <X11/Xwinsock.h>
 #endif
--- vnc_unixsrc/Xvnc/programs/Xserver/os/connection.c~	Mon May 11 19:02:17 1998
+++ vnc_unixsrc/Xvnc/programs/Xserver/os/connection.c	Mon Jul  1 01:05:07 2002
@@ -64,6 +64,8 @@
  *
  *****************************************************************/
 
+#define _GNU_SOURCE
+
 #ifdef WIN32
 #include <X11/Xwinsock.h>
 #endif
--- vnc_unixsrc/Xvnc/programs/Xserver/os/io.c~	Sat Jan 18 07:58:00 1997
+++ vnc_unixsrc/Xvnc/programs/Xserver/os/io.c	Mon Jul  1 01:05:24 2002
@@ -55,6 +55,8 @@
  *
  *****************************************************************/
 
+#define _GNU_SOURCE
+
 #ifdef WIN32
 #include <X11/Xwinsock.h>
 #endif
--- vnc_unixsrc/Xvnc/programs/Xserver/os/xdmcp.c~	Sat Jan 18 07:58:04 1997
+++ vnc_unixsrc/Xvnc/programs/Xserver/os/xdmcp.c	Mon Jul  1 01:07:10 2002
@@ -15,6 +15,8 @@
  *
  */
 
+#define _GNU_SOURCE
+
 #ifdef WIN32
 /* avoid conflicting definitions */
 #define BOOL wBOOL
--- vnc_unixsrc/Xvnc/lib/font/fc/fserve.c~	Wed Jun 11 14:08:41 1997
+++ vnc_unixsrc/Xvnc/lib/font/fc/fserve.c	Mon Jul  1 01:12:17 2002
@@ -53,6 +53,8 @@
  * font server specific font access
  */
 
+#define _GNU_SOURCE
+
 #ifdef WIN32
 #define _WILLWINSOCK_
 #endif
--- vnc_unixsrc/Xvnc/lib/font/fc/fsio.c~	Mon Mar  2 12:56:50 1998
+++ vnc_unixsrc/Xvnc/lib/font/fc/fsio.c	Mon Jul  1 01:12:33 2002
@@ -28,6 +28,8 @@
  * font server i/o routines
  */
 
+#define _GNU_SOURCE
+
 #ifdef WIN32
 #define _WILLWINSOCK_
 #endif
