--- xorg/mi/Makefile.am~	2007-09-06 13:40:39.000000000 +0200
+++ xorg/mi/Makefile.am	2007-09-06 13:41:26.567200032 +0200
@@ -67,3 +67,6 @@
 	mizerarc.h	\
 	mizerclip.c	\
 	mizerline.c
+
+libmi_la_LIBADD = $(XSERVERLIBS_LIBS)
+
--- xorg/hw/xnest/Makefile.am~	2007-09-06 13:40:39.000000000 +0200
+++ xorg/hw/xnest/Makefile.am	2007-09-06 13:41:58.057326203 +0200
@@ -53,6 +53,7 @@
 Xnest_LDADD = $(XORG_CORE_LIBS) \
               $(XNEST_LIBS) \
               $(XNESTMODULES_LIBS) \
+	      $(XSERVERLIBS_LIBS) \
 	      libfbcmap.a
 
 EXTRA_DIST = icon \
--- xorg/hw/dmx/dmxinit.c~	2007-08-23 21:04:52.000000000 +0200
+++ xorg/hw/dmx/dmxinit.c	2007-09-06 13:19:29.865279423 +0200
@@ -117,6 +117,17 @@
 
 Bool            dmxAddRemoveScreens = FALSE;
 
+int
+NewInputDeviceRequest(InputOption *options, DeviceIntPtr *pdev)
+{
+    return BadValue;
+}
+
+void
+DeleteInputDeviceRequest(DeviceIntPtr dev)
+{
+}
+
 /* dmxErrorHandler catches errors that occur when calling one of the
  * back-end servers.  Some of this code is based on _XPrintDefaultError
  * in xc/lib/X11/XlibInt.c */
--- xorg/hw/dmx/Makefile.am.org	2007-09-06 12:57:41.461829430 +0200
+++ xorg/hw/dmx/Makefile.am	2007-09-06 12:57:58.574980965 +0200
@@ -85,6 +85,7 @@
 Xdmx_LDADD = $(XORG_CORE_LIBS) \
              $(XDMX_LIBS) \
              $(GLX_LIBS) \
+	     $(XSERVERLIBS_LIBS) \
              input/libdmxinput.a \
              config/libdmxconfig.a \
              @DMXMODULES_LIBS@

From: Gerte Hoogewerf <g.hoogewerf@gmail.com>
Date: Tue, 4 Sep 2007 23:09:38 +0000 (-0700)
Subject: Add stub symbols to make xprint build.
X-Git-Url: http://gitweb.freedesktop.org/?p=xorg/xserver.git;a=commitdiff;h=d67e210f3458b62d7d4a6032aabfda0004d661c1

Add stub symbols to make xprint build.
---

--- a/hw/xprint/ddxInit.c
+++ b/hw/xprint/ddxInit.c
@@ -310,6 +310,17 @@ ChangeDeviceControl (
     return BadMatch;
 }
 
+int
+NewInputDeviceRequest(InputOption *options, DeviceIntPtr *pdev)
+{
+    return BadValue;
+}
+
+void
+DeleteInputDeviceRequest(DeviceIntPtr dev)
+{
+}
+
 void
 OpenInputDevice (
     DeviceIntPtr dev,
--- xorg-server-1.4/hw/xprint/Makefile.am~	2007-08-23 21:04:54.000000000 +0200
+++ xorg-server-1.4/hw/xprint/Makefile.am	2007-09-06 13:29:49.917161431 +0200
@@ -9,7 +9,7 @@
 	-DXFree86Server
 
 Xprt_LDFLAGS = -L$(top_srcdir)
-Xprt_LDADD = @XPRINT_LIBS@ ps/libps.la raster/libraster.la  \
+Xprt_LDADD = @XPRINT_LIBS@ @XSERVERLIBS_LIBS@ ps/libps.la raster/libraster.la  \
 	pcl/libpcl.la pcl-mono/libpcl.la $(top_builddir)/fb/libfb.la \
 	$(top_builddir)/render/librender.la $(top_builddir)/mi/libmi.la \
 	$(top_builddir)/Xext/libXext.la  @FREETYPE_LIBS@

